src/ppx_import.ml

```diff
@@ -465,79 +465,72 @@ let rec psig_of_tsig ~subst (tsig : Compat.signature_item_407 list) :
   | [] -> []
   | _ -> assert false
 
-let module_type ~tool_name ~input_name (modtype_decl : Ppxlib.module_type) =
+let module_type ~tool_name ~input_name
+    (package_type : Ppxlib__Import.package_type) =
   let open Ppxlib in
-  match modtype_decl with
-  | { pmty_attributes = _
-    ; pmty_desc = Pmty_extension ({txt = "import"; loc}, payload)
-    ; _ } -> (
-    match payload with
-    | PTyp {ptyp_desc = Ptyp_package (({txt = lid; loc} as alias), subst); _} ->
-      if tool_name = "ocamldep" then
-        if is_self_reference ~input_name lid then
-          (* Create a dummy module type to break the circular dependency *)
-          {modtype_decl with pmty_desc = Pmty_signature []}
-        else
-          (* Just put it as alias *)
-          {modtype_decl with pmty_desc = Pmty_alias alias}
-      else
-        Ppxlib.Ast_helper.with_default_loc loc (fun () ->
-            let env = Lazy.force lazy_env in
-            let tmodtype_decl =
-              match lid with
-              | Longident.Lapply _ ->
-                Location.raise_errorf ~loc
-                  "[%%import] cannot import a functor application %s"
-                  (string_of_lid lid)
-              | Longident.Lident _ as head_id ->
-                (* In this case, we know for sure that the user intends this lident
-                   as a module type name, so we use Typetexp.find_type and
-                   let the failure cases propagate to the user. *)
-                Compat.find_modtype env ~loc head_id |> snd
-              | Longident.Ldot (parent_id, elem) ->
-                let sig_items = locate_sig ~loc env parent_id in
-                get_modtype_decl ~loc sig_items parent_id elem
-            in
-            match tmodtype_decl with
-            | {mtd_type = Some (Mty_signature tsig); _} ->
-              let subst =
-                List.map (fun ({txt; _}, typ) -> (`Lid txt, typ)) subst
-              in
-              let psig =
-                psig_of_tsig ~subst
-                  (List.map Compat.migrate_signature_item tsig)
-              in
-              {modtype_decl with pmty_desc = Pmty_signature psig}
-            | {mtd_type = None; _} ->
-              Location.raise_errorf ~loc "Imported module is abstract"
-            | _ ->
-              Location.raise_errorf ~loc "Imported module is indirectly defined")
-    | _ -> Location.raise_errorf ~loc "Invalid [%%import] syntax" )
-  | _ -> modtype_decl
+  let ({txt = lid; loc} as alias), subst = package_type in
+  if tool_name = "ocamldep" then
+    if is_self_reference ~input_name lid then
+      (* Create a dummy module type to break the circular dependency *)
+      Ast_helper.Mty.mk ~attrs:[] (Pmty_signature [])
```

Comment:
That's one of the changes I was talking about. A new `module_type` is created from scratch here. We can't reuse any `attributes` from a `module_type` we receive as input anymore so I used an empty list. Could that cause any issues?